### PR TITLE
[Fabric] 2.x chaincode lifecycle update

### PR DESCRIPTION
**Pull request by himanshulalarya**
Signed-off-by: Lal <himanshu.a.lal@accenture.com>

**Changelog**
- Add Chaincode commit
- Updated Fabric 2.x lifecycle update

 

**Reviewed by**
@jagpreetsinghsasan 

 

**Linked issue**
#980

### CODE REVIEW

File: platforms/corda-ent/configuration/roles/setup/nmap/tasks/main.yml
Comment (sownak) [inline platforms/corda-ent/configuration/roles/setup/nmap/tasks/main.yml:2]: merging needs to be fixed

**Review by sownak — CHANGES_REQUESTED** [review]

File: platforms/hyperledger-fabric/charts/commit_chaincode/templates/_helpers.tpl
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/charts/commit_chaincode/templates/_helpers.tpl:5]: Add newline

File: platforms/hyperledger-fabric/charts/install_chaincode/templates/install_chaincode.yaml
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/charts/install_chaincode/templates/install_chaincode.yaml:54]: Make this variable from {{ VAR}} to {{ VAR }}

File: platforms/hyperledger-fabric/charts/install_chaincode/templates/install_chaincode.yaml
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/charts/install_chaincode/templates/install_chaincode.yaml:171]: change this to echo "Chaincode installed for Fabric 2.x"

File: platforms/hyperledger-fabric/charts/install_chaincode/templates/install_chaincode.yaml
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/charts/install_chaincode/templates/install_chaincode.yaml:165]: Change this echo to something meaningful

File: platforms/hyperledger-fabric/charts/install_chaincode/templates/install_chaincode.yaml
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/charts/install_chaincode/templates/install_chaincode.yaml:159]: Change this echo to something meaningful

File: platforms/hyperledger-fabric/charts/install_chaincode/templates/install_chaincode.yaml
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/charts/install_chaincode/templates/install_chaincode.yaml:160]: Remove this

File: platforms/hyperledger-fabric/charts/install_chaincode/templates/install_chaincode.yaml
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/charts/install_chaincode/templates/install_chaincode.yaml:163]: Change this to echo "Chaincode install for Fabric v.1.4.4"

File: platforms/hyperledger-fabric/configuration/roles/create/chaincode/commit/tasks/main.yaml
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/roles/create/chaincode/commit/tasks/main.yaml:16]: Add newline

File: platforms/hyperledger-fabric/configuration/roles/create/orderers/vars/main.yaml
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/roles/create/orderers/vars/main.yaml:10]: Add newline

File: platforms/hyperledger-fabric/configuration/roles/helm_component/templates/commit_chaincode_job.tpl
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/roles/helm_component/templates/commit_chaincode_job.tpl:44]: Add newline

File: platforms/hyperledger-fabric/configuration/samples/network-fabric-add-organization.yaml
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabric-add-organization.yaml:424]: Dont put nodejs as an option until we have it

**Review by jagpreetsinghsasan — CHANGES_REQUESTED** [review]

**Comment by jagpreetsinghsasan** [comment]
Make sure to always squash and merge all commits into 1

**Review by jagpreetsinghsasan — APPROVED** [review]
Just make sure to have end-to-end testing before this gets merged. The code seems fine now.

File: platforms/hyperledger-fabric/charts/invoke_chaincode/templates/invoke_chaincode.yaml
Comment (sownak) [inline platforms/hyperledger-fabric/charts/invoke_chaincode/templates/invoke_chaincode.yaml:130]: why hardcoded to 1.4.4?

File: platforms/hyperledger-fabric/configuration/roles/delete/flux_releases/meta/main.yaml
Comment (sownak) [inline platforms/hyperledger-fabric/configuration/roles/delete/flux_releases/meta/main.yaml:6]: need to be removed

File: platforms/hyperledger-fabric/configuration/roles/helm_component/templates/install_chaincode_job.tpl
Comment (sownak) [inline platforms/hyperledger-fabric/configuration/roles/helm_component/templates/install_chaincode_job.tpl:44]: if this fails when lang does not exist in network.yaml, then that needs to be corrected.

**Review by sownak — CHANGES_REQUESTED** [review]

**Comment by sownak** [comment]
@jagpreetsinghsasan Please resolve conversation after checking the updates

**Review by jagpreetsinghsasan — APPROVED** [review]

**Review by sownak — APPROVED** [review]

**Comment by sownak** [comment]
Approved, but will be merged after current release.

**Pull request merged**
